src/box/txn.h

```diff
@@ -893,6 +893,38 @@ txn_is_nop(const struct txn *txn)
 	return txn->n_new_rows + txn->n_applier_rows == 0;
 }
 
+static inline bool
+txn_is_fully_temporary(struct txn *txn)
+{
+	if (!txn_is_nop(txn))
+		return false;
+	struct txn_stmt *stmt;
+	stailq_foreach_entry(stmt, &txn->stmts, next) {
+		if (stmt->space != NULL && !space_is_temporary(stmt->space) &&
+		    space_is_data_temporary(stmt->space))
+			return false;
+	}
+	return true;
+}
+
+static inline bool
+txn_is_fully_remote(struct txn *txn)
+{
+	if (txn->n_new_rows != 0)
+		return false;
+	struct txn_stmt *stmt;
+	stailq_foreach_entry(stmt, &txn->stmts, next) {
+		/*
+		 * Allow DDL on data-temporary spaces, since we allow only fully
+		 * temporary transactions to continue.
+		 */
+		if (stmt->row == NULL && stmt->space != NULL &&
```

Comment:
Nit: the function is just a checker, looks strange that it mentions DDL. I guess this comments rather belongs to the place, where this function is used?

Reply:
This function determines the filtering condition. This comment explains why we take filter out only statements with `stmt->row == NULL`.

Reply:
AFAIU, you expect that replace in a data-temporary space has `stmt->row == NULL` and DDL there has `stmt->row != NULL`. Well, any DDL is actually a replace in space `_space` which is not temporary at all, so it still would be allowed without this condition. Please, drop it.

Reply:
@drewdzzz Thanks, moved the `stmt->row == NULL` to an assert and moved the comment out of the loop.